esmvalcore/config-developer.yml

```diff
@@ -236,7 +236,7 @@ CORDEX:
     spec: '{domain}/{institute}/{driver}/{exp}/{ensemble}/{dataset}/{rcm_version}/{mip}/{short_name}'
     BADC: '{domain}/{institute}/{driver}/{exp}/{ensemble}/{dataset}/{rcm_version}/{mip}/{short_name}/{latestversion}'
   input_file: '{short_name}_{domain}_{driver}_{exp}_{ensemble}_{dataset}_{rcm_version}_{mip}*.nc'
-  output_file: '{short_name}_{dataset}_{exp}_{ensemble}_{rcm_version}_{mip}'
+  output_file: '{short_name}_{domain}_{driver}_{dataset}_{exp}_{ensemble}_{rcm_version}_{mip}'
```

Comment:
Adding these facets looks fine to me, but @zklaus why did we choose such a different scheme for this output file name than for the other projects? Most of the time we start with something like '{project}_{dataset}_{mip}_{exp}_{ensemble}_{short_name}'.

Reply:
Maybe. For a global domain indicate only the dataset should be enogh. But for regionals domains in most of the applications it is required the domain (for example EUR-11 stands for Europe at 11km, AFR-22 Africa at 22km) and is also required the driver, which is the name of the global model used as input for the regional model.

Reply:
So something like `{project}_{domain}_{driver}_{dataset}_{mip}_{exp}_{ensemble}_{rcm_version}_{short_name}`? Just the format we use for CMIP6 with the extra tags in (I hope) sensible locations. It will look more consistent with all the other data we produce

Reply:
@francesco-cmcc My apologies for not being clear enough, I was referring to the ordering of the facets.
@jvegasbsc Yes, I think that is more in line with the others, but maybe move domain a but further back because it's not very generic and I would definitely put dataset (and possibly rcm_version) before drive. @zklaus Since you implemented the original of this, do you have an opinion on the order?

Reply:
@francesco-cmcc @zklaus any further opinions on this? Would be nice to include this in v2.5!

Reply:
The order suggested by @jvegreg and @bouweandela is fine by me. Originally I followed the order used by CMIP and CORDEX, but that's a discussion for a different day.

@francesco-cmcc, could you update this PR accordingly?